### PR TITLE
Get the supported storage tiers of a cloud instance

### PR DESCRIPTION
**Pull request by kishen-v**
Changes:
* Introduce a storage tier client to retrieve the storage tiers available to a associated cloud instance ID.

### CODE REVIEW

**Comment by kishen-v** [comment]
cc @yussufsh

**Review by yussufsh — DISMISSED** [review]
lgtm

File: clients/instance/ibm-pi-storage-tiers.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+// NewIBMPIStorageTierClient
+func NewIBMPIStorageTierClient(ctx context.Context, sess *ibmpisession.IBMPISession, cloudInstanceID string) *IBMPIStorageTierClient {
+	return &IBMPIStorageTierClient{
+		*NewIBMPIClient(ctx, sess, cloudInstanceID),
+	}
+}
+
+// Gets all the storage tiers associated to the cloud instance.
+func (f *IBMPIStorageTierClient) GetAll(id string) (models.RegionStorageTiers, error) {
```
Comment (michaelkad) [inline clients/instance/ibm-pi-storage-tiers.go:28]:
```suggestion
func (f *IBMPIStorageTierClient) GetAll() (models.RegionStorageTiers, error) {
```

File: clients/instance/ibm-pi-storage-tiers.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	params := p_cloud_storage_tiers.NewPcloudCloudinstancesStoragetiersGetallParams().
+		WithContext(f.ctx).WithTimeout(helpers.PIGetTimeOut).
+		WithCloudInstanceID(id)
```
Comment (michaelkad) [inline clients/instance/ibm-pi-storage-tiers.go:31]:
```suggestion
		WithCloudInstanceID(f.cloudInstanceID)
```

**Review by michaelkad — CHANGES_REQUESTED** [review]
No need to pass the `id`

**Comment by kishen-v** [comment]
Hi @michaelkad, thanks for reviewing the PR!

I have made the suggested changes. PTAL when time permits.

**Review by michaelkad — APPROVED** [review]

**Pull request merged**